torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -598,3 +598,81 @@ def _update_schema_suggestion_for_cat(
         )
     ]
     return output_sharding
+
+
+@register_prop_rule(aten.split.Tensor)
+def split_rule(op_schema: OpSchema) -> OutputSharding:
+    output_spec_list: List[DTensorSpec] = []
+    input_spec = cast(DTensorSpec, op_schema.args_schema[0])
+    ndim = input_spec.ndim
+    split_size_or_sections = op_schema.args_schema[1]
+    dim = (
+        cast(int, op_schema.args_schema[2])
+        if len(op_schema.args_schema) > 2
+        else 0
+    )
+    dim = normalize_dim(dim, ndim)
+
+    # TODO: tensor to split cannot have _Partial
+    # in its placements for now. Will need to
+    # support in future.
+    if input_spec.sums:
+        raise NotImplementedError(
+            f"splitting distributed tensor with "
+            f"_Partial placement is not implemented!\n"
+            f"DTensorSpec={input_spec}"
+        )
+
+    # TODO: just like slice op, split replicates before
+    # splitting on a sharded dimension
+    need_reshard = False
+    if is_tensor_dim_sharded(input_spec, dim=dim):
```

Comment:
This somehow broke TP's code logic. Because a common technique people are using is that the DTensor is sharded on the last dim and they call split on the last dim too. We still want the result to be sharded on dim=-1.

Reply:
Currently after split, we got replicate as a DTensor.